### PR TITLE
Skip dataAvailabilityMode in program output

### PR DESCRIPTION
**Pull request by adamiak**
This is a fix for Apex Goerli testnet, which switches to a different program output (one more field) at block 8056029.

### CODE REVIEW

File: packages/encoding/src/decoding/decodePerpetualCairoOutput.ts [hunk 1]
```diff
@@ -8,9 +8,16 @@ import { readForcedActions } from './readForcedActions'
...
 
-export function decodePerpetualCairoOutput(data: string): PerpetualCairoOutput {
+export function decodePerpetualCairoOutput(
+  data: string,
+  skipDataAvailabilityModeField?: boolean
+): PerpetualCairoOutput {
   const reader = new ByteReader(data)
 
+  if (skipDataAvailabilityModeField) {
+    const _dataAvailabilityMode = reader.readBigInt(32)
```
Comment (torztomasz) [inline packages/encoding/src/decoding/decodePerpetualCairoOutput.ts:18]:
```suggestion
    reader.readBigInt(32)
```

File: packages/backend/src/core/collectors/PerpetualCairoOutputCollector.ts [hunk 2]
```diff
@@ -13,16 +14,31 @@ const ABI = new utils.Interface([
...
     }
     const decoded = ABI.decodeFunctionData('updateState', tx.data)
     const programOutputValues = decoded.programOutput as BigNumber[]
     const hexData = toHexData(programOutputValues)
-    return decodePerpetualCairoOutput(hexData)
+
+    // This is a fix for Apex Goerli testnet, which switches
+    // to a different program output (one more field) at block 8056029.
+    // See: https://github.com/starkware-libs/stark-perpetual/blob/eaa284683deec190407fece98b96546d10f6ad67/src/services/perpetual/cairo/output/program_output.cairo#L36
+    const skipDataAvailabilityModeField =
```
Comment (torztomasz) [inline packages/backend/src/core/collectors/PerpetualCairoOutputCollector.ts:37]: I would extract it to separate method called `shouldSkipDataAvailabilityModeField` so it does not make mess in collect function. The comment could be moved there also.

**Review by torztomasz — CHANGES_REQUESTED** [review]

File: packages/backend/src/core/collectors/PerpetualCairoOutputCollector.ts (hunk 2 above)
Comment (adamiak) [reply packages/backend/src/core/collectors/PerpetualCairoOutputCollector.ts]: Done

File: packages/backend/src/core/collectors/shouldSkipDataAvailabilityModeField.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { InstanceName } from '@explorer/shared'
+
+export function shouldSkipDataAvailabilityModeField(
```
Comment (torztomasz) [inline packages/backend/src/core/collectors/shouldSkipDataAvailabilityModeField.ts:3]: In my opinion this function should be inside PerpetualCairoOutputCollector.ts

**Review by torztomasz — APPROVED** [review]

**Pull request merged**